internal/wal_segment_runner.go

```diff
@@ -0,0 +1,114 @@
+package internal
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/wal-g/storages/storage"
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/utility"
+)
+
+type WalSegmentNotFoundError struct {
+	error
+}
+
+func newWalSegmentNotFoundError(segmentFileName string) WalSegmentNotFoundError {
+	return WalSegmentNotFoundError{
+		errors.Errorf("Segment file '%s' does not exist in storage.\n", segmentFileName)}
+}
+
+func (err WalSegmentNotFoundError) Error() string {
+	return fmt.Sprintf(tracelog.GetErrorFormatter(), err.error)
+}
+
+type ReachedStopSegmentError struct {
+	error
+}
+
+func newReachedStopSegmentError() ReachedStopSegmentError {
+	return ReachedStopSegmentError{errors.Errorf("Reached stop segment.\n")}
+}
+
+func (err ReachedStopSegmentError) Error() string {
+	return fmt.Sprintf(tracelog.GetErrorFormatter(), err.error)
+}
+
+type WalSegmentDescription struct {
+	number   WalSegmentNo
+	timeline uint32
+}
+
+func (desc *WalSegmentDescription) GetFileName() string {
+	return desc.number.getFilename(desc.timeline)
+}
+
+// WalSegmentRunner is used for sequential iteration over WAL segments in the storage
+type WalSegmentRunner struct {
+	currentWalSegment WalSegmentDescription
+	walFolderSegments map[WalSegmentDescription]bool
+	stopSegmentNo     WalSegmentNo
+}
+
+func NewWalSegmentRunner(
+	startWalSegment WalSegmentDescription,
+	segments map[WalSegmentDescription]bool,
+	stopSegmentNo WalSegmentNo,
+) *WalSegmentRunner {
+	return &WalSegmentRunner{startWalSegment,
+		segments, stopSegmentNo}
+}
+
+// MoveNext tries to get the next segment from storage
+func (r *WalSegmentRunner) MoveNext() (WalSegmentDescription, error) {
+	if r.currentWalSegment.number <= r.stopSegmentNo {
+		return WalSegmentDescription{}, newReachedStopSegmentError()
+	}
+	nextSegment := r.getNextSegment()
+	var fileExists bool
```

Comment:
You don't use it in other places. Replace "var" definition with ":=" definition.

Reply:
Fixed in 07b35b1